managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -2287,5 +2287,15 @@ public String getState() {
         return STATE_UPDATER.get(this).toString();
     }
 
+    @Override
+    public double getThrottleMarkDelete() {
+        return this.markDeleteLimiter.getRate();
+    }
+
+    @Override
+    public void setThrottleMarkDelete(double throttleMarkDelete) {
+        this.markDeleteLimiter.setRate(throttleMarkDelete);
```

Comment:
`markDeleteLimiter` can also be `null` if the rate was initially 0. Also, the limiter instance is final. 

I would suggest to remove the `final` qualifier and reset to null or from null in cases where the rate is or was 0, otherwise use the `setRate()`. Do not worry about making it `volatile` since this is not critical to get picked up immediately

Reply:
yes, also missed to handle NPE.